src/audio/Kconfig

```diff
@@ -561,6 +561,12 @@ config PCM_CONVERTER_FORMAT_S24LE
 	help
 	  Support 24 bit processing data format with sign and in little endian format
 
+config PCM_CONVERTER_FORMAT_S24BE
+	bool "Support S24_BE"
+	default y
+	help
+	  Support 24 bit processing data format with sign and in big endian format
+
```

Comment:
I like the cleanup of the pcm converter, but I don't think we should call this "S24_BE". Unless I'm badly mistaken, this has nothing to do with endianess, which is the standard meaning for LE/BE shorthands. 

I.e. we always use little-endian encoding, i.e. 24bit number 0x00aabbcc is stored in memory as "cc bb aa 00". 

The LSB_INTEGER/MSB_INTEGER controls whether the encoding is "cc bb aa 00" or "00 cc bb aa" (order of bytes is not changed).

If we had S24_BE, the ALSA interpretation for this is to encode the same number as "00 aa bb cc". S32_BE would be "aa bb cc 00" (order of bytes changes). And so forth.

Reply:
@kv2019i  Thanks for review!  I copy the comments in ALSA pcm.h.   And if I want to support MSB_INTEGER like "cc bb aa 00", do you suggest to use S32_BE ?

````
// For S32/U32 formats, 'msbits' hardware parameter is often used to deliver information about the
// available bit count in most significant bit. It's for the case of so-called 'left-justified' or
// `right-padding` sample which has less width than 32 bit.
````

Reply:
Definitely not S32_BE, the BE is about endianess (order of bytes) while this about left/right-justified like the ALSA documentation states.

In ALSA, the MSB_INTEGER is just S32_LE (with msbits=24). If you interpret "cc bb aa 00" as S32_LE, it just works, there is no need for additional format in most cases.

But alas, we have not followed this in "enum sof_ipc_frame", so it's ugly to add back now. 

Maybe one option is to have "S24_4LE_MSB" which matches 100% the ALSA definition of "S32_LE with msbit=24". Then we'd have mapping from ALSA to SOF:
    - SOF_S24_4LE_MSB => S32_LE with msbits=24
    - SOF_S24_4LE          => S24_LE 
    - SOF_S24_3LE          => S24_3LE
    - SOF_S32_LE            => S32_LE with msbits=32

Makes sense?

Reply:
sure, thanks! Good idea!